testnet/stacks-node/src/config.rs

```diff
@@ -1409,18 +1409,21 @@ impl BurnchainConfigFile {
                     // Using std::net::LookupHost would be preferable, but it's
                     // unfortunately unstable at this point.
                     // https://doc.rust-lang.org/1.6.0/std/net/struct.LookupHost.html
-                    let mut sock_addrs = format!("{}:1", &peer_host)
-                        .to_socket_addrs()
-                        .map_err(|e| format!("Invalid burnchain.peer_host: {}", &e))?;
-                    let sock_addr = match sock_addrs.next() {
-                        Some(addr) => addr,
-                        None => {
+                    let mut attempts = 0;
+                    let mut addrs_iter = loop {
+                        if let Ok(addrs_iter) = format!("{}:1", peer_host).to_socket_addrs() {
+                            break addrs_iter;
+                        }
+                        attempts += 1;
+                        if attempts == 15 {
                             return Err(format!(
                                 "No IP address could be queried for '{}'",
                                 &peer_host
                             ));
                         }
+                        sleep(std::time::Duration::from_secs(5));
                     };
+                    let sock_addr = addrs_iter.next().unwrap();
```

Comment:
IMO in non-test code it's better to use `.expect()` with an error message. Maybe something like "Failed to connect to peer"?